spec/unit/pops/types/type_calculator_spec.rb

```diff
@@ -832,10 +885,40 @@ class Foo
         tested_types = all_types - [
           Puppet::Pops::Types::PAnyType,
           Puppet::Pops::Types::POptionalType,
+          Puppet::Pops::Types::PNotUndefType,
           Puppet::Pops::Types::PDataType] - collection_types
         t = Puppet::Pops::Types::PStructType.new()
         tested_types.each {|t2| t.should_not be_assignable_to(t2.new) }
       end
+
+      it 'Default key optionality is controlled by value assignability to undef' do
```

Comment:
Not sure I agree. The test verifies that if the value is not assignable to undef, then the key is not wrapped in an Optional and if the value is assignable to undef, the key is wrapped in an Optional. I.e. "the key optionality is controlled by value assignability to undef". Can you find a better short and concise wording for it?